Index: include/llvm/Target/TargetCallingConv.h
===================================================================
--- include/llvm/Target/TargetCallingConv.h	(revision 129634)
+++ include/llvm/Target/TargetCallingConv.h	(working copy)
@@ -36,6 +36,8 @@
     static const uint64_t ByValAlignOffs = 6;
     static const uint64_t Split          = 1ULL << 10;
     static const uint64_t SplitOffs      = 10;
+    static const uint64_t SplitPiece     = 0x3FULL << 11;
+    static const uint64_t SplitPieceOffs = 11;
     static const uint64_t OrigAlign      = 0x1FULL<<27;
     static const uint64_t OrigAlignOffs  = 27;
     static const uint64_t ByValSize      = 0xffffffffULL << 32; //< Struct size
@@ -93,6 +95,13 @@
       Flags = (Flags & ~ByValSize) | (uint64_t(S) << ByValSizeOffs);
     }
 
+    unsigned getSplitPiece() const {
+      return (unsigned)((Flags & SplitPiece) >> SplitPieceOffs);
+    }
+    void setSplitPiece(unsigned S) {
+      Flags = (Flags & ~SplitPiece) | (uint64_t(S) << SplitPieceOffs);
+    }
+
     /// getArgFlagsString - Returns the flags as a string, eg: "zext align:4".
     std::string getArgFlagsString();
 
