Index: include/llvm/ADT/Triple.h
===================================================================
--- include/llvm/ADT/Triple.h	(revision 129685)
+++ include/llvm/ADT/Triple.h	(working copy)
@@ -48,6 +48,7 @@
 
     alpha,   // Alpha: alpha
     arm,     // ARM; arm, armv.*, xscale
+    avr,     // AVR: avr
     bfin,    // Blackfin: bfin
     cellspu, // CellSPU: spu, cellspu
     mips,    // MIPS: mips, mipsallegrex
@@ -71,6 +72,7 @@
   enum VendorType {
     UnknownVendor,
 
+    Atmel,
     Apple,
     PC,
     SCEI
@@ -83,6 +85,7 @@
     Darwin,
     DragonFly,
     FreeBSD,
+    FreeRTOS,
     Linux,
     Lv2,        // PS3
     MinGW32,    // i*86-pc-mingw32, *-w64-mingw32
Index: lib/Support/Triple.cpp
===================================================================
--- lib/Support/Triple.cpp	(revision 129685)
+++ lib/Support/Triple.cpp	(working copy)
@@ -25,6 +25,7 @@
 
   case alpha:   return "alpha";
   case arm:     return "arm";
+  case avr:     return "avr";
   case bfin:    return "bfin";
   case cellspu: return "cellspu";
   case mips:    return "mips";
@@ -57,6 +58,8 @@
   case arm:
   case thumb:   return "arm";
 
+  case avr:     return "avr";
+
   case bfin:    return "bfin";
 
   case cellspu: return "spu";
@@ -83,6 +86,7 @@
   case UnknownVendor: return "unknown";
 
   case Apple: return "apple";
+  case Atmel: return "atmel";
   case PC: return "pc";
   case SCEI: return "scei";
   }
@@ -99,6 +103,7 @@
   case Darwin: return "darwin";
   case DragonFly: return "dragonfly";
   case FreeBSD: return "freebsd";
+  case FreeRTOS: return "freertos";
   case Linux: return "linux";
   case Lv2: return "lv2";
   case MinGW32: return "mingw32";
@@ -131,6 +136,8 @@
     return alpha;
   if (Name == "arm")
     return arm;
+  if (Name == "avr")
+    return avr;
   if (Name == "bfin")
     return bfin;
   if (Name == "cellspu")
@@ -225,6 +232,8 @@
     return "ppc64";
   if (Str == "mblaze" || Str == "microblaze")
     return "mblaze";
+  if (Str == "avr")
+    return "avr";
   if (Str == "arm")
     return "arm";
   if (Str == "armv4t" || Str == "thumbv4t")
@@ -258,6 +267,8 @@
     return ppc64;
   else if (ArchName == "mblaze")
     return mblaze;
+  else if (ArchName == "avr")
+    return avr;
   else if (ArchName == "arm" ||
            ArchName.startswith("armv") ||
            ArchName == "xscale")
@@ -295,6 +306,8 @@
 Triple::VendorType Triple::ParseVendor(StringRef VendorName) {
   if (VendorName == "apple")
     return Apple;
+  if (VendorName == "atmel")
+    return Atmel;
   else if (VendorName == "pc")
     return PC;
   else if (VendorName == "scei")
@@ -312,6 +325,8 @@
     return Darwin;
   else if (OSName.startswith("dragonfly"))
     return DragonFly;
+  else if (OSName.startswith("freertos"))
+    return FreeRTOS;
   else if (OSName.startswith("freebsd"))
     return FreeBSD;
   else if (OSName.startswith("linux"))
