Index: include/clang/Basic/Attr.td
===================================================================
--- include/clang/Basic/Attr.td (revisión: 151992)
+++ include/clang/Basic/Attr.td (copia de trabajo)
@@ -336,6 +336,14 @@
   let Spellings = [];
 }

+def AVRInterruptHandler : InheritableAttr {
+  let Spellings = [];
+}
+
+def AVRInterrupt : InheritableAttr {
+  let Spellings = ["interrupt"];
+}
+
 def Naked : InheritableAttr {
   let Spellings = ["naked"];
 } 
