Index: cmake/config-ix.cmake
===================================================================
--- cmake/config-ix.cmake	(revision 84974)
+++ cmake/config-ix.cmake	(working copy)
@@ -106,6 +106,8 @@
   set(LLVM_NATIVE_ARCH Alpha)
 elseif (LLVM_NATIVE_ARCH MATCHES "arm")
   set(LLVM_NATIVE_ARCH ARM)
+elseif (LLVM_NATIVE_ARCH MATCHES "avr")
+  set(LLVM_NATIVE_ARCH AVR)
 elseif (LLVM_NATIVE_ARCH MATCHES "mips")
   set(LLVM_NATIVE_ARCH Mips)
 elseif (LLVM_NATIVE_ARCH MATCHES "pic16")
