Index: lib/CodeGen/AsmPrinter/AsmPrinter.cpp
===================================================================
--- lib/CodeGen/AsmPrinter/AsmPrinter.cpp       (revisión: 169957)
+++ lib/CodeGen/AsmPrinter/AsmPrinter.cpp       (copia de trabajo)
@@ -287,7 +287,10 @@
   unsigned AlignLog = getGVAlignmentLog2(GV, *TD);

   // Handle common and BSS local symbols (.lcomm).
-  if (GVKind.isCommon() || GVKind.isBSSLocal()) {
+  // AVR Specific: symbols that do not belong to the generic address space need
+  // to be placed in a different section and not use the comm directive.
+  if ((GVKind.isCommon() || GVKind.isBSSLocal())
+      && GV->getType()->getAddressSpace() == 0) {
     if (Size == 0) Size = 1;   // .comm Foo, 0 is undefined, avoid it.
     unsigned Align = 1 << AlignLog;
 
