Index: AsmWriter.cpp
===================================================================
--- AsmWriter.cpp	(revisión: 152005)
+++ AsmWriter.cpp	(copia de trabajo)
@@ -1505,6 +1505,7 @@
   case CallingConv::ARM_AAPCS:    Out << "arm_aapcscc "; break;
   case CallingConv::ARM_AAPCS_VFP:Out << "arm_aapcs_vfpcc "; break;
   case CallingConv::MSP430_INTR:  Out << "msp430_intrcc "; break;
+  case CallingConv::AVR_INTR:     Out << "avr_intrcc "; break;
   case CallingConv::PTX_Kernel:   Out << "ptx_kernel "; break;
   case CallingConv::PTX_Device:   Out << "ptx_device "; break;
   default: Out << "cc" << F->getCallingConv() << " "; break;
@@ -1806,6 +1807,7 @@
     case CallingConv::ARM_AAPCS:    Out << " arm_aapcscc "; break;
     case CallingConv::ARM_AAPCS_VFP:Out << " arm_aapcs_vfpcc "; break;
     case CallingConv::MSP430_INTR:  Out << " msp430_intrcc "; break;
+    case CallingConv::AVR_INTR:     Out << " avr_intrcc "; break;
     case CallingConv::PTX_Kernel:   Out << " ptx_kernel"; break;
     case CallingConv::PTX_Device:   Out << " ptx_device"; break;
     default: Out << " cc" << CI->getCallingConv(); break;
@@ -1862,6 +1864,7 @@
     case CallingConv::ARM_AAPCS:    Out << " arm_aapcscc "; break;
     case CallingConv::ARM_AAPCS_VFP:Out << " arm_aapcs_vfpcc "; break;
     case CallingConv::MSP430_INTR:  Out << " msp430_intrcc "; break;
+    case CallingConv::AVR_INTR:     Out << " avr_intrcc "; break;
     case CallingConv::PTX_Kernel:   Out << " ptx_kernel"; break;
     case CallingConv::PTX_Device:   Out << " ptx_device"; break;
     default: Out << " cc" << II->getCallingConv(); break;
