Index: CallingConv.h
===================================================================
--- CallingConv.h	(revisión: 152005)
+++ CallingConv.h	(copia de trabajo)
@@ -94,7 +94,10 @@
 
     /// MBLAZE_INTR - Calling convention used for MBlaze interrupt support
     /// routines (i.e. GCC's save_volatiles attribute).
-    MBLAZE_SVOL = 74
+    MBLAZE_SVOL = 74,
+
+    /// AVR_INTR - Calling convention used for AVR interrupt routines.
+    AVR_INTR = 75
   };
 } // End CallingConv namespace
 
