Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 151472)
+++ CMakeLists.txt	(working copy)
@@ -72,6 +72,7 @@
 
 set(LLVM_ALL_TARGETS
   ARM
+  AVR
   CBackend
   CellSPU
   CppBackend
Index: cmake/config-ix.cmake
===================================================================
--- cmake/config-ix.cmake	(revision 151472)
+++ cmake/config-ix.cmake	(working copy)
@@ -315,6 +315,8 @@
   set(LLVM_NATIVE_ARCH PowerPC)
 elseif (LLVM_NATIVE_ARCH MATCHES "arm")
   set(LLVM_NATIVE_ARCH ARM)
+elseif (LLVM_NATIVE_ARCH MATCHES "avr")
+  set(LLVM_NATIVE_ARCH AVR)
 elseif (LLVM_NATIVE_ARCH MATCHES "mips")
   set(LLVM_NATIVE_ARCH Mips)
 elseif (LLVM_NATIVE_ARCH MATCHES "xcore")
