Index: include/llvm/CallingConv.h
===================================================================
--- include/llvm/CallingConv.h  (revisión: 165770)
+++ include/llvm/CallingConv.h  (copia de trabajo)
@@ -112,7 +112,13 @@
     /// Cannot have variable arguments.
     /// Can also be called by the host.
     /// Is externally visible.
-    SPIR_KERNEL = 76
+    SPIR_KERNEL = 76,
+
+    /// AVR_INTR - Calling convention used for AVR interrupt routines.
+    AVR_INTR = 77,
+
+    /// AVR_SIGNAL - Calling convention used for AVR signal routines.
+    AVR_SIGNAL = 78
   };
 } // End CallingConv namespace
 
Index: lib/VMCore/AsmWriter.cpp
===================================================================
--- lib/VMCore/AsmWriter.cpp    (revisión: 165770)
+++ lib/VMCore/AsmWriter.cpp    (copia de trabajo)
@@ -78,6 +78,8 @@
     case CallingConv::ARM_AAPCS:    Out << "arm_aapcscc"; break;
     case CallingConv::ARM_AAPCS_VFP:Out << "arm_aapcs_vfpcc"; break;
     case CallingConv::MSP430_INTR:  Out << "msp430_intrcc"; break;
+    case CallingConv::AVR_INTR:     Out << "avr_intrcc "; break;
+    case CallingConv::AVR_SIGNAL:   Out << "avr_signalcc "; break;
     case CallingConv::PTX_Kernel:   Out << "ptx_kernel"; break;
     case CallingConv::PTX_Device:   Out << "ptx_device"; break;
     default:                        Out << "cc" << cc; break;
Index: lib/AsmParser/LLParser.cpp
===================================================================
--- lib/AsmParser/LLParser.cpp  (revisión: 165770)
+++ lib/AsmParser/LLParser.cpp  (copia de trabajo)
@@ -1100,6 +1100,8 @@
 ///   ::= 'arm_aapcscc'
 ///   ::= 'arm_aapcs_vfpcc'
 ///   ::= 'msp430_intrcc'
+///   ::= 'avr_intrcc'
+///   ::= 'avr_signalcc'
 ///   ::= 'ptx_kernel'
 ///   ::= 'ptx_device'
 ///   ::= 'spir_func'
@@ -1119,6 +1121,8 @@
   case lltok::kw_arm_aapcscc:    CC = CallingConv::ARM_AAPCS; break;
   case lltok::kw_arm_aapcs_vfpcc:CC = CallingConv::ARM_AAPCS_VFP; break;
   case lltok::kw_msp430_intrcc:  CC = CallingConv::MSP430_INTR; break;
+  case lltok::kw_avr_intrcc:     CC = CallingConv::AVR_INTR; break;
+  case lltok::kw_avr_signalcc:   CC = CallingConv::AVR_SIGNAL; break;
   case lltok::kw_ptx_kernel:     CC = CallingConv::PTX_Kernel; break;
   case lltok::kw_ptx_device:     CC = CallingConv::PTX_Device; break;
   case lltok::kw_spir_kernel:    CC = CallingConv::SPIR_KERNEL; break;
Index: lib/AsmParser/LLLexer.cpp
===================================================================
--- lib/AsmParser/LLLexer.cpp   (revisión: 165770)
+++ lib/AsmParser/LLLexer.cpp   (copia de trabajo)
@@ -523,6 +523,8 @@
   KEYWORD(arm_aapcscc);
   KEYWORD(arm_aapcs_vfpcc);
   KEYWORD(msp430_intrcc);
+  KEYWORD(avr_intrcc);
+  KEYWORD(avr_signalcc);
   KEYWORD(ptx_kernel);
   KEYWORD(ptx_device);
   KEYWORD(spir_kernel);
Index: lib/AsmParser/LLToken.h
===================================================================
--- lib/AsmParser/LLToken.h     (revisión: 165770)
+++ lib/AsmParser/LLToken.h     (copia de trabajo)
@@ -80,6 +80,8 @@
     kw_x86_stdcallcc, kw_x86_fastcallcc, kw_x86_thiscallcc,
     kw_arm_apcscc, kw_arm_aapcscc, kw_arm_aapcs_vfpcc,
     kw_msp430_intrcc,
+    kw_avr_intrcc,
+    kw_avr_signalcc,
     kw_ptx_kernel, kw_ptx_device,
     kw_spir_kernel, kw_spir_func,
