Index: include/llvm/ADT/Triple.h
===================================================================
--- include/llvm/ADT/Triple.h	(revision 90055)
+++ include/llvm/ADT/Triple.h	(working copy)
@@ -57,6 +57,7 @@
     
     alpha,   // Alpha: alpha
     arm,     // ARM; arm, armv.*, xscale
+    avr,     // AVR: avr
     bfin,    // Blackfin: bfin
     cellspu, // CellSPU: spu, cellspu
     mips,    // MIPS: mips, mipsallegrex
Index: lib/Support/Triple.cpp
===================================================================
--- lib/Support/Triple.cpp	(revision 90055)
+++ lib/Support/Triple.cpp	(working copy)
@@ -24,6 +24,7 @@
     
   case alpha:   return "alpha";
   case arm:     return "arm";
+  case avr:     return "avr";
   case bfin:    return "bfin";
   case cellspu: return "cellspu";
   case mips:    return "mips";
@@ -109,6 +110,8 @@
     return alpha;
   if (Name == "arm")
     return arm;
+  if (Name == "avr")
+    return avr;
   if (Name == "bfin")
     return bfin;
   if (Name == "cellspu")
@@ -226,6 +229,8 @@
     Arch = bfin;
   else if (ArchName == "pic16")
     Arch = pic16;
+  else if (ArchName == "avr")
+    Arch = avr;
   else if (ArchName == "powerpc")
     Arch = ppc;
   else if ((ArchName == "powerpc64") || (ArchName == "ppu"))
