Index: include/llvm/ADT/Triple.h
===================================================================
--- include/llvm/ADT/Triple.h	(revision 151472)
+++ include/llvm/ADT/Triple.h	(working copy)
@@ -44,6 +44,7 @@
     UnknownArch,
 
     arm,     // ARM; arm, armv.*, xscale
+    avr,     //
     cellspu, // CellSPU: spu, cellspu
     hexagon, // Hexagon: hexagon
     mips,    // MIPS: mips, mipsallegrex
Index: lib/Support/Triple.cpp
===================================================================
--- lib/Support/Triple.cpp	(revision 151472)
+++ lib/Support/Triple.cpp	(working copy)
@@ -20,6 +20,7 @@
   case UnknownArch: return "unknown";
 
   case arm:     return "arm";
+  case avr:     return "avr";
   case cellspu: return "cellspu";
   case hexagon: return "hexagon";
   case mips:    return "mips";
@@ -54,6 +55,8 @@
   case arm:
   case thumb:   return "arm";
 
+  case avr:   return "avr";
+
   case cellspu: return "spu";
 
   case ppc64:
@@ -135,6 +138,7 @@
 Triple::ArchType Triple::getArchTypeForLLVMName(StringRef Name) {
   return StringSwitch<Triple::ArchType>(Name)
     .Case("arm", arm)
+    .Case("avr", avr)
     .Case("cellspu", cellspu)
     .Case("mips", mips)
     .Case("mipsel", mipsel)
@@ -201,6 +205,7 @@
     .Case("powerpc", "ppc")
     .Case("powerpc64", "ppc64")
     .Cases("mblaze", "microblaze", "mblaze")
+    .Case("avr", "avr")
     .Case("arm", "arm")
     .Cases("armv4t", "thumbv4t", "armv4t")
     .Cases("armv5", "armv5e", "thumbv5", "thumbv5e", "armv5")
@@ -243,6 +248,7 @@
     .Case("ptx64", Triple::ptx64)
     .Case("le32", Triple::le32)
     .Case("amdil", Triple::amdil)
+    .Case("avr", Triple::avr)
     .Default(Triple::UnknownArch);
 }
 
@@ -629,6 +635,7 @@
     return 0;
 
   case llvm::Triple::msp430:
+  case llvm::Triple::avr:
     return 16;
 
   case llvm::Triple::amdil:
@@ -676,6 +683,7 @@
   switch (getArch()) {
   case Triple::UnknownArch:
   case Triple::msp430:
+  case Triple::avr:
     T.setArch(UnknownArch);
     break;
 
@@ -713,6 +721,7 @@
   case Triple::UnknownArch:
   case Triple::amdil:
   case Triple::arm:
+  case Triple::avr:
   case Triple::cellspu:
   case Triple::hexagon:
   case Triple::le32:
